wandb/sdk/launch/github_reference.py

```diff
@@ -138,7 +142,10 @@ def parse(uri: str) -> Optional["GitHubReference"]:
                         repo = repo[: -len(SUFFIX_GIT)]
                     ref.repo = repo
                     ref.view = parts[3] if len(parts) > 3 else None
-                    ref.path = "/".join(parts[4:])
+                    if len(parts) > 4:
+                        ref.ref = parts[4]
+                        ref.ref_type = ReferenceType.BRANCH
```

Comment:
We need to consider some edge cases - in GitHub, you could have a branch "jamie/test" but you could also have a directory structure "jamie/test" and until we have it cloned we can't really guess.